lymph/unilateral.py

```diff
@@ -960,10 +963,15 @@ def assign_parameters(self, **kwargs):
         """
 
         for key, value in kwargs.items():
-            if key[:5] == 'micro':
-                self.parameters[key].micro_mod = value
+            if key in self.diag_time_dists:
+                try:
+                    self.diag_time_dists[key].update(value)
+                except ValueError as val_err:
+                    raise ValueError(
+                        "Parameters for marginalization over diagnose times are invalid"
+                        ) from val_err
             else:
-                self.parameters[key].spread_prob = value
+                self._setter_lookup[key](value)

```

Comment:
Nice! Now that I look at this: We could implement a subclass of `dict` to handle this. Then we could write
```python
self._setter_lookup[key] = value
```
and the dict-subclass handles our custom logic. Nothing urgent though 😉

Reply:
Maybe I'll implement that at a later point